### DISCUSSION TITLE
Request for comments for SIG-Governance

### DISCUSSION THREAD

**Discussion by OBWANDO** [o3de/community#105] [General]
# **O3DE SIG-Governance**

## **SIG Chairperson Elections**

Every O3DE SIG must elect a Chairperson. Chairpersons are elected to serve for a one-year term. There are no term limits for SIG Chairpersons.

### Election officers

One or two O3DE community volunteers serve as election officers of the SIG Chairperson elections. The O3DE TSC, Executive Director or Program Manager will put out a call for community volunteers to administer the election process. The first two volunteers from different organizations will be approved to serve as the election officers. 

Election officer responsibilities:

* Create election folders on Github under each SIG’s repository where candidates will declare their nomination, interest and platform.
* Keep the election process on track and on time for all of the SIGs
* Send a call for nominations & follow-up reminders to the SIG Discord channels
* Administrate the election voting tool
* Announce the election results to the O3DE community 

### Election dates

The SIG Chairperson elections take place once every year. The election process begins three calendar weeks prior to the election.

* One week to call for volunteers to serve as Election Officers
* One week for SIG Chairperson candidates to submit their candidacy
* One week for community voting on all SIG Chairperson elections simultaneously
* Results announced

### Electorate 

Contributors (defined as anyone who has contributed by pull request into the O3DE project in the last 12 months) are eligible to vote in the SIG Chairperson elections.

### Candidacy

Anyone who has made contributions to the O3DE project is eligible to run. Candidates must nominate themselves and submit their candidacy by adding a text file to the corresponding election folder under the SIG repository with their interest and platform, and announcing their candidacy on the SIG’s Discord channel. There are no term limits for SIG Chairpersons.

### Voting Tool

SIGs may use a voting tool that shows public results like Doodle or a similar tool.

### Uncontested Elections

In the event that there is only one nominee for the SIG Chairperson position, the election is considered uncontested, and therefore the one nominee will be confirmed as the new SIG Chairperson without the need for voting. 

### Initial Organization tasks

After the inception of the O3DE project in 2021, the initial Chairperson elected for each SIG will be responsible to work with the SIG community for finalizing the official Charter document of their specific SIG. 

Each SIG has an official Charter documented in it’s Github repository. Sections of the Charter that need to be completed by the Chairperson include: 

* Overview of the SIG
* Goals 
* Scope
* SIG Links & Lists

SIG Charters:

* [SIG-Build](https://github.com/o3de/sig-build/blob/main/governance/SIG%20Build%20Charter.md)
* [SIG-Content](https://github.com/o3de/sig-content/blob/main/governance/SIG%20Content%20Creation%20Charter.md)
* [SIG-Core](https://github.com/o3de/sig-core/blob/main/governance/SIG%20Core%20Charter.md)
* [SIG-Docs](https://github.com/o3de/sig-docs-community/blob/main/governance/charter.md)
* [SIG-Network](https://github.com/o3de/sig-network/blob/main/governance/SIG%20Network%20Charter.md)
* [SIG-Operations](https://github.com/o3de/sig-operations/blob/main/governance/charter.md)
* [SIG-Platform](https://github.com/o3de/sig-platform/blob/main/governance/SIG%20Platform%20Charter.md)
* [SIG-Graphics-Audio](https://github.com/o3de/sig-graphics-audio/blob/main/governance/SIG-Graphics-AudioCharter.md)
* [SIG-Release](https://github.com/o3de/sig-release/blob/main/governance/SIG%20Release%20Charter.md)
* [SIG-Security](https://github.com/o3de/sig-security/blob/main/governance/SIG%20Security%20Charter.md)
* [SIG-Testing](https://github.com/o3de/sig-testing/blob/main/governance/SIG%20Testing%20Charter.md)
* [SIG-UI-UX](https://github.com/o3de/sig-ui-ux/blob/main/governance/SIG%20UI-UX%20Charter.md)

### Delegation responsibilities

Delegating is a large part of your role as a SIG Chairperson as it is the primary approach to scale with the community. There are numerous tasks and it can be difficult to keep up with it all. Some SIGs are more fortunate than others, having multiple people around to delegate to, however this is not always the case - no matter the size of the SIG.

The following are some tips and tricks to help you delegate:

* Reach out to SIG members on Discord or Github discussions - everyone communicates differently.
* Detail your task needing to be fulfilled. Vague requests tend to go ignored because people have their own workloads. But if you need someone to host a SIG meeting or even fix a bug, details are key to getting results.
* Don’t wait until the last minute to ask for help. 

If you can’t find a delegate, it is okay to let things go. It is not the utmost importance to have a SIG meeting, or plan everything perfectly. Here are some tips to help you deal with being unable to delegate tasks:

* Do not be afraid to reach out to other SIGs or the TSC for help. The TSC is designed to provide guidance and support where possible.
* Don’t be a hero, burnout is real. Ensure people are aware that you are having troubles and some deliverables might not be met. We care about our community members, and it’s important that you feel supported, and not crushed.

### Recurrent tasks

The recurrent tasks outlined below are a minimum expectation that the community believes the serving SIG Chairperson should be capable of as the serving team leader. Keep in mind that the items listed below are optional and dependent entirely on your SIG’s function.

1. Schedule the recurring public SIG meetings and publish the meeting invite on the [O3DE community calendar](https://lists.o3de.org/g/o3de-calendar/calendar).
2. Organize the SIG meeting agendas and publish them under the SIG’s repository on Github.
3. Keep track of the SIG related O3DE development cycle tasks (unless you have appointed a SIG-Release liaison to follow it for you)
4. Ensure regular scheduling and operation of triaging incoming issues
5. Facilitate all SIG communications in the open using the public Discord channels (voice and text), Github and mailing lists
6. Ensure that SIG team members adhere to the [Contributor Covenant Code of Conduct](https://github.com/o3de/community/blob/main/code-of-conduct.md)
7. Lead by example and promote the [Open 3D Community Values](https://github.com/o3de/community/blob/main/values.md)

### Reviewer & Maintainer maintenance

1. Familiarize yourself with the [Reviewer and Maintainer criteria](https://github.com/o3de/community/blob/main/community-membership.md) 
2. Encourage the promotion of community members to Reviewers and active reviewers to Maintainers
3. Check in regularly with Maintainers who have become inactive
4. When a Reviewer or Maintainer no longer meets the criteria requirements, initiate the process to demote them

### At the beginning of a new O3DE release cycle

1. Identify and present features that are to be developed for the next release to TSC.
2. Check if the SIG meeting time works for most active contributors.
3. If you keep release liaison responsibilities, join the #SIG-Release channel in Discord and be sure to follow release discussions in Github
4. Delegate responsibilities to track removed and deprecated features
5. If the TSC has approved community goals, check the relevance of the goal to the SIG and assess work items for the team.

### During the O3DE release cycle

1. Help first time contributors, they will be struggling the most.
2. Work with other SIGs Chairs for cross cutting issues.
3. Lack of reviews? Reach out to the SIG maintainers and remind them.

### At the end of the O3DE release cycle

1. Coordinate with SIG-Release and SIG-Docs for cleanup and publishing of release notes.
2. Coordinate with the SIG-Release for deliverables, unless a liaison has been appointed.
3. Perform a retrospective with the SIG. Suggested sections include: What went well, What didn’t go well.
4. Ensure documentation is up-to-date with any major changes that were implemented throughout the cycle.

### Collecting feedback

Collecting feedback from users and operators is an essential step for incrementally improving software. Anyone can collect feedback, but sometimes it falls on the shoulders of the SIG Chairperson to facilitate open lines of communication.

**Comment by lmbr-pip** [comment]
See that guide is missing how SIGs can remove ineffective leadership during the year. If we have SIG leads who are inactive, or are actively alienating to the community, should there be a process that covers removal before their 12 month term at the helm is up?

Most of whats above looks great and carefully written thought, some general thoughts

**Candidacy**
"Anyone who has made contributions to the O3DE project is eligible to run. Candidates must nominate themselves and submit their candidacy by adding a text file to the corresponding election folder under the SIG repository with their interest and platform, and announcing their candidacy on the SIG’s Discord channel. There are no term limits for SIG Chairpersons." 
- It great that anyone can run and be nominated and anyone can vote. I personally feel there has to be a balance around prior engagement in that SIG, that potential leaders of SIGs have had some active contributions in the SIG area (ie PR, RFC, Maintainer, PR reviews) otherwise it runs into danger of being a popularity contest and SIGs end up with leaders without any relevant domain knowledge.

**Voting Tool**
"SIGs may use a voting tool that shows public results like Doodle or a similar tool" 
- IMHO we should be consistent with voting tools so folks don't need to learn x different voting tools / register in tools or are somehow excluded (accessibility, platform access etc) by SIGs choice. Would also state that SIGs need to publically post results and  archive results  for public record keeping/transparency.

**At the beginning of a new O3DE release cycle**
"Identify and present features that are to be developed for the next release to TSC."
- Not sure if this mechanism really exists today, so will need to be fleshed out esp if there is contention or guidance down from the TSC.

**During the O3DE release cycle**
"Work with other SIGs Chairs for cross cutting issues" 
- As a general point cross-cutting across multiple SIGs probably needs process help. Its can be extremely time consuming to engage with multiple SIGs on cross-cutting issues. It asks a lot of people with limited time. Having process help would really help.
